presto-main/src/test/java/com/facebook/presto/operator/scalar/TestZipTransformFunction.java

```diff
@@ -0,0 +1,94 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.type.ArrayType;
+import com.facebook.presto.type.RowType;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.facebook.presto.spi.type.VarcharType.createVarcharType;
+import static com.facebook.presto.type.UnknownType.UNKNOWN;
+import static java.util.Arrays.asList;
+
+public class TestZipTransformFunction
+        extends AbstractTestFunctions
+{
+    @Test
+    public void testSameLength()
+            throws Exception
+    {
+        assertFunction("zip(ARRAY[1, 2], ARRAY['a', 'b'], (x, y) -> (y, x))",
+                zipReturnType(createVarcharType(1), INTEGER),
+                list(list("a", 1), list("b", 2)));
+
+        assertFunction("zip(ARRAY[1, 2], ARRAY[CAST('a' AS VARCHAR), CAST('b' AS VARCHAR)], (x, y) -> (y, x))",
+                zipReturnType(VARCHAR, INTEGER),
+                list(list("a", 1), list("b", 2)));
+
+        assertFunction("zip(ARRAY[], ARRAY[], (x, y) -> (y, x))",
+                zipReturnType(UNKNOWN, UNKNOWN),
+                list());
+
+        assertFunction("zip(ARRAY[NULL], ARRAY[NULL], (x, y) -> (y, x))",
+                zipReturnType(UNKNOWN, UNKNOWN),
+                list(list(null, null)));
+
+        assertFunction("zip(ARRAY[1, 1], ARRAY[1, 2], (x, y) -> x + y)",
+                new ArrayType(INTEGER),
+                list(2, 3));
+    }
+
+    @Test
+    public void testDifferentLength()
+            throws Exception
+    {
+        assertFunction("zip(ARRAY[1], ARRAY['a', 'b'], (x, y) -> (y, x))",
+                zipReturnType(createVarcharType(1), INTEGER),
+                list(list("a", 1), list("b", null)));
+
+        assertFunction("zip(ARRAY[NULL, 2], ARRAY['a'], (x, y) -> (y, x))",
+                zipReturnType(createVarcharType(1), INTEGER),
+                list(list("a", null), list(null, 2)));
+
+        assertFunction("zip(ARRAY[1, NULL], ARRAY[3, 2, 1], (x, y) -> x + y)",
+                new ArrayType(INTEGER),
+                list(4, null, null));
+    }
+
+    @Test
+    public void testWithNull()
+            throws Exception
+    {
+        assertFunction("zip(CAST(NULL AS ARRAY(UNKNOWN)), ARRAY[], (x, y) -> (y, x))",
+                zipReturnType(UNKNOWN, UNKNOWN),
+                null);
+    }
+
+    private static Type zipReturnType(Type... types)
```

Comment:
I would inline this function. 
* The name as is does not properly convey what this function does.
* A proper name for this function is hard.
* The function appears to be simple enough to just inline.